P2022/tests/header/3way.hpp

```diff
@@ -1,16 +1,32 @@
 
 
 #include <iterator>
+#include <concepts>
 
 namespace std::ranges
 {
+    template<
+        input_iterator I1,
+        input_iterator I2,
+        class Comp,
+        class Proj1,
+        class Proj2
+    >
+    using lexicographical_compare_three_way_result_t =
+        invoke_result_t<Comp, invoke_result_t<Proj1, I1>, invoke_result_t<Proj2, I2>>;
+
     template<
         input_iterator I1, sentinel_for<I1> S1,
         input_iterator I2, sentinel_for<I2> S2,
         class Comp = compare_three_way,
         class Proj1 = identity, 
         class Proj2 = identity
     >
+    requires
+        std::convertible_to<lexicographical_compare_three_way_result_t<I1,I2,Comp,Proj1,Proj2>, strong_ordering> ||
```

Comment:
This is not the right way to go , convertible to is not expressive of what you want it to be, i think same_as is better